controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -1180,13 +1180,18 @@ void releaseStepLocks(AbstractOperationContext.Step step) {
                 try {
                     modelController.awaitContainerStability(timeout, TimeUnit.MILLISECONDS, true);
                 }  catch (InterruptedException e) {
+                    // Cancelled in some way
                     interrupted = true;
-                    MGMT_OP_LOGGER.interruptedWaitingStability();
+                    // Cancellation prevented us ensuring that MSC will be stable when this operation completes.
+                    // We can no longer have any sense of MSC state or how the model relates to the runtime and
+                    // we need to start from a fresh service container. Notify the controller of this.
+                    modelController.containerCannotStabilize();
```

Comment:
Should there be some kind of differentiation between this interrupted case, and the timeout case below?